conftest.py

```diff
@@ -176,37 +168,61 @@ def function_to_seed_torch(seed):
 
 
 @pytest.fixture(autouse=True)
-def autoseeding_of_everything(record_property, request):
+def autoseeding_of_everything(request):
     """Seed everything we can, for determinism."""
-    main_seed = request.config.getoption("--forcing_random_seed", default=None)
+    randomly_seed = request.config.getoption("--randomly-seed", default=None)
 
-    if main_seed is None:
-        main_seed = random.randint(0, 2**64 - 1)
+    if randomly_seed is None:
+        raise ValueError("why isn't --randomly-seed set?")
 
-    seed = main_seed
-    record_property("main seed", main_seed)
+    # We need to find the relative file path of the test file. It does not look native with request,
+    # so we recompute it.
+    absolute_path = str(request.fspath)
+
+    assert (
+        absolute_path.count("concrete-ml/test") == 1
+    ), f"the absolute path is not as expected {absolute_path=}"
+
+    relative_file_path = absolute_path[
+        absolute_path.find("concrete-ml/test") + len("concrete-ml/") :
+    ]
+
+    # Derive the sub_seed from the randomly_seed and the test name
+    derivation_string = f"{relative_file_path} # {str(request.node.name)} # {randomly_seed}"
+
+    hash_object = hashlib.sha256()
+    hash_object.update(b"{derivation_string}")
+    hash_object.digest()
+    hash_value = hash_object.hexdigest()
+
+    sub_seed = int(hash_value, 16) % 2**64
+
+    print(f"\nUsing {randomly_seed=}\nUsing {derivation_string=}\nUsing {sub_seed=}")
+
+    # And then, do everything per this sub_seed
+    seed = sub_seed
 
-    # Python
-    random.seed(seed)
-    print("\nForcing seed to random.seed to ", seed)
     print(
-        f"\nRelaunch the tests with --forcing_random_seed {seed} "
-        + "--randomly-dont-reset-seed to reproduce. Remark that adding --randomly-seed=... "
-        + "is needed when the testcase uses randoms in pytest parameters"
+        f"\nRelaunch the tests with --randomly_seed {randomly_seed} "
```

Comment:
repro directly a subtest by:
- forcing --randomly_seed
- setting the subtest you want
and that's it

Reply:
what do you mean by "setting the subtest you want" ? isn't running `pytest tests/common/test_utils.py::test_compute_bits_precision[True-False-5] --randomly_seed 78975873 --randomly-dont-reset-seed` enough for example ?

Reply:
Now with the changes, yes `pytest tests/common/test_utils.py::test_compute_bits_precision[True-False-5] --randomly_seed 78975873 --randomly-dont-reset-see` is enough but that's exactly the change. Before, it was NOT sufficient, we have to use `--forcing_random_seed ` option

Reply:
Tried to re-explain what it does in this thread

Reply:
okok got it you were commenting the actual workflow, not what users was supposed to do my bad !